styleguide.md

```diff
@@ -0,0 +1,207 @@
+
+### The keywords themselves
+
+When referencing a keyword itself, hyphenation is not used, and when possible, a better describing word than _keyword_ is used:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+the `input` prefix | `\lstinline!prefix! prefix` | For more prefixes, see `class-prefixes`, `base-prefix`, and `type-prefix` in the grammar.
+the `final` modifier | `the \lstinline!final! modifier` |
+the `each` keyword | `the \lstinline!each! keyword` |
+
+Depending on context, one can also swap the order, or drop the describing word completely:
+
+Appearance | LaTeX source
+--- | ---
+the keyword `each` | `the keyword \lstinline!each!`
+declared as `final` | `declared as \lstinline!final!`
+
+### Named functions and operators
+
+When referencing a named function of operator, hyphenation is not used, and it is common to not combine with any describing word:
+
+Appearance | LaTeX source
+--- | ---
+the `smooth` operator | `the \lstinline!smooth! operator`
+where `pre` is not explicitly used | `where \lstinline!pre! is not explicitly used`
+
+### Other things, special cases
+
+Incomplete list of various terminology with special formatting rules:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+start value | `start value` | Value of the `start`-attribute (there could be exceptions!)
+reduction expression | `reduction expression` |
+base class | `base class` | Similarly: derived class
+base-clock | `base-clock` | Similarly: sub-cock
+
+## Definitions
+
+New terminology is either introduced with a `definition` environment, or as part of the running text.
+When part of the running text, the introduced terminology should be marked with `\firstuse` at the point of the definition.
+As a general rule, `\firstuse` should appear in combination with `\index` for adding the term to the document index.
+(The use of `\firstuse` instead of just `\emph` helps us both produce consistent formatting and makes it easier to spot cases where the additional use of `\index` has been forgotten.
+The reason that `\firstuse` doesn't also do the job calling `\index` is that the form of the term presented to `\firstuse` isn't always in the base form expected in the document index, that there can be a need for special styling tricks in the `\index` argument, etc.)
+
+If the new terminology is used before being introduced, it should be marked with `\willintroduce` (instead of `\firstuse`) to alert the reader that this is not a term that is expected to be known yet by a first-time reader.
+
+## Miscellaneous
+
+### Use of `\emph` and italics
+
+To put emphasis on a word or small piece of text, use `\emph`.
+
+Italics is used when new terminology is introduced in the running text instead of the bulkier `definition` environment, see `\firstuse` and `\willintroduce`.
+
+Non-semantical font switching commands for producing italics (`\textit`, `\textsl`, `\itshape`) should pretty much never be used.
```

Comment:
"... should pretty much never used" seems like an awkward phrase. Why not something simple as
"Refrain from using non-semantical font switching commands for producing italics (...).